### PR TITLE
Optimization 3825 v1

### PR DESCRIPTION
**Pull request by jufajardini**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3825

Describe changes:
- filecontainer: add Files structure, to replace/unify SMBFiles and NFSFiles
- smb/files: delete SMBFiles implementation
- smb/smb: replace SMBFiles with Files
- nfs/nfs: delete NFSFiles implementation, replace former declarations with Files' ones

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by jufajardini** [comment]
So, taking into account the commit guidelines, and one of the first feedbacks Shivani offered, I wasn't sure, here, how many commits I should have, nor what should be the best way of logically organising them. 

Should they be done by file - as in, should I have 4 commits for this? Or could it be like I did, by type of change and directories involved?

Thanks in advance for any feedbacks!

**Comment by jufajardini** [comment]
I will close it because there's a missing module error, but I still appreciate some feedback regarding the previous comment, if possible :)

File: rust/src/smb/smb.rs [hunk 1]
```diff
@@ -808,7 +809,7 @@ impl SMBState {
             ssn2vecoffset_map:HashMap::new(),
             ssn2tree_map:HashMap::new(),
             ssnguid2vec_map:HashMap::new(),
-            files: SMBFiles::new(),
+            files: Files::new(), // replacing SMBFiles with Files
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:812]: this comment can be removed

**Review by victorjulien — CHANGES_REQUESTED** [review]
I think the change is small enough to be a single commit. It also helps the issue that each commit needs to build correctly.

Can you do a new branch + new PR with a single commit?

File: rust/src/filecontainer.rs [hunk 2]
```diff
@@ -25,6 +25,39 @@ extern {
     pub fn FileFlowToFlags(flow: *const Flow, flags: u8) -> u16;
 }
 
+// Unifying the definition of files SMBFiles and NFSFiles (HTTP2 too, soon)
```
Comment (victorjulien) [inline rust/src/filecontainer.rs:28]: this comment should be updated to what it is now, it should not be about what it was or how we got there. That type of info should go into the git commit messages.

File: rust/src/smb/smb.rs [hunk 3]
```diff
@@ -755,7 +755,8 @@ pub struct SMBState<> {
     // requests for DCERPC.
     pub ssnguid2vec_map: HashMap<SMBHashKeyHdrGuid, Vec<u8>>,
 
-    pub files: SMBFiles,
+    // replacing SMBFiles with Files
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:758]: this comment should be removed

File: rust/src/nfs/nfs.rs [hunk 4]
```diff
@@ -315,7 +283,8 @@ pub struct NFSState {
     /// transactions list
     pub transactions: Vec<NFSTransaction>,
 
-    pub files: NFSFiles,
+    // replacing NFSFiles with the new structure derived from it
```
Comment (victorjulien) [inline rust/src/nfs/nfs.rs:286]: comment should be removed

**Comment by jufajardini** [comment]
Changes implemented in new PR: https://github.com/OISF/suricata/pull/5513

**Pull request closed without merging**